src/jsTest/kotlin/io/github/oshai/kotlinlogging/SimpleJsTest.kt

```diff
@@ -20,41 +32,165 @@ class SimpleJsTest {
+  @Test
+  fun checkTopLevelLambdaLoggerName() {
+    checkLoggerName(topLevelLambdaLogger, "SimpleJsTest")
+  }
+
+  @Test
+  fun checkTopLevelLambdaLoggerInfoMessage() {
+    checkLoggerInfoMessage(topLevelLambdaLogger)
+  }
+
+  @Test
+  fun checkTopLevelLambdaLoggerErrorMessage() {
+    checkLoggerErrorMessage(topLevelLambdaLogger)
+  }
+
+  @Test
+  fun checkTopLevelLambdaLoggerOffLevel() {
+    checkLoggerOffLevel(topLevelLambdaLogger)
+  }
+
+  // ClassNamedLogger
+  @Test
+  fun checkClassNamedLoggerName() {
+    checkLoggerName(MyClass().classNamedLogger, "MyClass")
+  }
+
   @Test
-  fun simpleJsTest() {
-    assertEquals("SimpleJsTest", logger.name)
+  fun checkClassNamedLoggerInfoMessage() {
+    checkLoggerInfoMessage(MyClass().classNamedLogger)
+  }
+
+  @Test
+  fun checkClassNamedLoggerErrorMessage() {
+    checkLoggerErrorMessage(MyClass().classNamedLogger)
+  }
+
+  @Test
+  fun checkClassNamedLoggerOffLevel() {
+    checkLoggerOffLevel(MyClass().classNamedLogger)
+  }
+
+  // ClassLambdaLogger
+  @Test
+  fun checkClassLambdaLoggerName() {
+    checkLoggerName(MyClass().classLambdaLogger, "MyClass")
+  }
+
+  @Test
+  fun checkClassLambdaLoggerInfoMessage() {
+    checkLoggerInfoMessage(MyClass().classLambdaLogger)
+  }
+
+  @Test
+  fun checkClassLambdaLoggerErrorMessage() {
+    checkLoggerErrorMessage(MyClass().classLambdaLogger)
+  }
+
+  @Test
+  fun checkClassLambdaLoggerOffLevel() {
+    checkLoggerOffLevel(MyClass().classLambdaLogger)
+  }
+
+  // CompanionNamedLogger
+  @Test
+  fun checkCompanionNamedLoggerName() {
+    checkLoggerName(MyClass.MyCompanion.companionNamedLogger, "MyClassCompanion")
+  }
+
+  @Test
+  fun checkCompanionNamedLoggerInfoMessage() {
+    checkLoggerInfoMessage(MyClass.MyCompanion.companionNamedLogger)
+  }
+
+  @Test
+  fun checkCompanionNamedLoggerErrorMessage() {
+    checkLoggerErrorMessage(MyClass.MyCompanion.companionNamedLogger)
+  }
+
+  @Test
+  fun checkCompanionNamedLoggerOffLevel() {
+    checkLoggerOffLevel(MyClass.MyCompanion.companionNamedLogger)
+  }
+
+  // CompanionLambdaLogger
+  @Test
+  fun checkCompanionLambdaLoggerName() {
+    checkLoggerName(MyClass.MyCompanion.companionLambdaLogger, "MyClass")
+  }
+
+  @Test
+  fun checkCompanionLambdaLoggerInfoMessage() {
+    checkLoggerInfoMessage(MyClass.MyCompanion.companionLambdaLogger)
+  }
+
+  @Test
+  fun checkCompanionLambdaLoggerErrorMessage() {
+    checkLoggerErrorMessage(MyClass.MyCompanion.companionLambdaLogger)
+  }
+
+  @Test
+  fun checkCompanionLambdaLoggerOffLevel() {
+    checkLoggerOffLevel(MyClass.MyCompanion.companionLambdaLogger)
+  }
+
+  // use cases
+  private fun checkLoggerName(logger: KLogger, expected: String) {
+    assertEquals(expected, logger.name)
+  }
+
+  private fun checkLoggerInfoMessage(logger: KLogger) {
     logger.info { "info msg" }
-    assertEquals("INFO: [SimpleJsTest] info msg", appender.lastMessage)
+    assertEquals("INFO: [${logger.name}] info msg", appender.lastMessage)
```

Comment:
can you change this to an explicit string param with logger name, so it will be clear what was the name?

Reply:
It's checking via `checkLoggerName(logger: KLogger, expected: String)`

Reply:
@oshai Hi, could we move this PR forward?